iree/turbine/aot/support/ir_utils.py

```diff
@@ -324,7 +324,8 @@ def create_tensor_global(
                 )
             else:
                 # Emit inline initialized.
-                contents = torch.utils.dlpack.to_dlpack(t)
+                detached_tensor = t.detach().contiguous().cpu()
```

Comment:
What is the memory impact here? Because detaching is kind of unclear to me. 

From what I saw in the code, PyTorch as a framework drops ownership of the data and it copies some/all of it somehow. What are the memory impacts here? Can you add a test to our internal suite. 

Could we guard it? Like
```suggestion
                detached_tensor = t
                if not detached_tensor.is_contiguous():
                    detached_tensor = detached_tensor.contiguous()
```
And I don't know about using the `.cpu()` call here. I would expect the model is some kind of FakeTensor/meta device during/after FxGraph export.

Reply:
That's the line they originally used in turbine. According to [https://pytorch.org/docs/stable/generated/torch.Tensor.detach.html] no copy is performed. And regarding the .cpu(): Can we be sure that the tensor is always on the cpu and the correct device? Because that's what .cpu() does, right? ([https://pytorch.org/docs/stable/generated/torch.Tensor.cpu.html])

Reply:
Yeah but I'm not 100% sure the original line is the right thing to do. DLPack is just a view on data. It does not take ownership so detaching semantically does not make sense. 

And for the cpu() case, we know nothing about where we are at this point in the code. I would not change it unless it is strictly required.

Reply:
Discussed Offline: this solution seems to give the best result in terms of memory,

Reply:
Yeah, running  the pytest has an leak of only 400B